### ISSUE TITLE
[java] ConsecutiveAppendsShouldReuse false-negative

### ISSUE THREAD

**Issue by boris-petrov** [pmd/pmd#1051]
This rule seems to work sporadically. I have cases with tens of lines of:

```java
builder.append(...);
builder.append(...);
builder.append(...);
```

Which don't give the warning. Not sure when it happens - perhaps when they are in a lambda where `builder` is a parameter.

**Comment by jsotuyod** [comment]
@boris-petrov thanks for the report. Can you provide a working sample we can analyze?

**Comment by boris-petrov** [comment]
@jsotuyod - well, for example the following code doesn't trigger the warning:

```java
public static void asd(StringBuilder builder, String b) {
        builder.append("asd");
        builder.append(b);
        builder.append(123);
}
```

**Comment by jsotuyod** [comment]
awesome, thanks! I can reproduce the issue with your snippet

**Comment by boris-petrov** [comment]
@jsotuyod - glad to hear it! :) Now that I think about it, that should be considered a `true-negative` rather than `false-negative`. `false-negative` meas that it is not an issue and PMD doesn't give a warning (which is correct). `true-negative` means that it **is** an issue but PMD doesn't give a warning. Am I right? :)

**Comment by oowekyala** [comment]
@boris-petrov No, `false-negative` means that PMD reports nothing (negative) and that this behaviour is wrong (false), ie it should report something. `false-X` are bugs, `true-X` are expected behaviour

Truly, this is a false negative ^^

**Comment by eli-mcgowan** [comment]
There are two cases where this can occur.
The first is a simplification of the snippet provided, it only requires that the StringBuilder be a method parameter.
The second is when the StringBuilder is a class variable.

(I have the fix prepared and will submit in the next 24 hours)

**Comment by jsotuyod** [comment]
@boris-petrov that's actually a different scenario. You would need the lambda type inference to detect that properly. If you replaced that with:

```java
stream.reduce(new StringBuilder(1_000), (StringBuilder builder, String string) -> {
	switch (string) {
		case "asd":
			builder.append("asd");
			builder.append("qwe");
			break;
	}
}
```

6.3.0 and older wouldn't report it, 6.4.0 will. I'll open a new issue to track this.

**Comment by jsotuyod** [comment]
Tracked as #1160
